### PR TITLE
Add ldap support

### PR DESCRIPTION
**Pull request by kay0u**
If you have a better idea than what I've done in  `scripts/actions/public_private`, tell me :-)

### CODE REVIEW

File: scripts/install [hunk 1]
```diff
@@ -119,6 +119,12 @@ ynh_replace_string "__ENCRYPT__" "$always_encrypt" "$final_path/lutim.conf"
 secret=$(ynh_string_random)
 ynh_replace_string "__SECRET__" "$secret" "$final_path/lutim.conf"
 ynh_app_setting_set $app secret "$secret"
+if [ $is_public -eq 0 ];
+then
+	ynh_replace_string "__IS_PUBLIC__" "" "$final_path/lutim.conf"
+else
+	ynh_replace_string "__IS_PUBLIC__" "#" "$final_path/lutim.conf"
+fi
```
Comment (maniackcrudelis) [inline scripts/install]: That's the opposite, if is_public is false, then you comment those line.

**Comment by maniackcrudelis** [comment]
It feels like it's the same than public/private.
If private, only a registered user in YunoHost can load images. If public everyone can.
Adding this feature limit users as well for public access. Not sure it worth the pain.

If it was a internal authentication to lutim, we could find a use case for users other than YunoHost ones. But as it's ldap authentication, it does the same thing than YunoHost, without HTTP Auth.

Is there any use case I missed ?

**Comment by kay0u** [comment]
The fact is, when ldap is activated via lutim, every user can manage his own images. If is not, I saw that it's web browser based (I believe)

File: scripts/install (hunk 1 above)
Comment (kay0u) [reply scripts/install]: No, the idea is:

If it's public, no login is asked (ldap on lutim side is not activated), if it's private, then login is asked, and users can manage there images.

**Comment by maniackcrudelis** [comment]
So... what was the purpose of adding ldap here ?
Having lutim more secured of having that feature for users that can manage their pictures ?

**Comment by kay0u** [comment]
The idea is to link the uploading of an image to a specific account. (Maybe this is not an interesting feature and you can close this PR:-))

**Comment by maniackcrudelis** [comment]
Well, no that can prove to be useful.
But, first that should be an option, since it adds an extra authentication field. It could be bothering if not needed.

Second, did you try it with different browsers to be sure that's indeed a different behavior ?

**Comment by kay0u** [comment]
Hmm in private navigation, different windows use same cookies... So my bad, ldap on lutim side don't link images to an account. With a different browser, images disappear.

**Comment by maniackcrudelis** [comment]
That's too bad because it could have been interesting...

**Comment by kay0u** [comment]
Yep... Sorry for the fake joy. We can close this PR then?

**Comment by maniackcrudelis** [comment]
I guess so...
Is there any other advantages to add ldap into lutim ?

I think the main purpose of the feature was only to restrain who can upload images.

**Comment by kay0u** [comment]
Yes, but it's already provided by yunohost.

**Pull request closed without merging**